### PR TITLE
fix: config consistency refactor

### PR DESCRIPTION
**Pull request by devthejo**

### CODE REVIEW

**Review by revu-bot-dev — COMMENTED** [review]
### Overview
This refactor centralizes runtime configuration and wires it into server startup. The direction is solid, but there are a few issues that could cause runtime failures and configuration inconsistencies.

| File | Lines | Severity | Issue | Action |
|---|---:|---|---|---|
| src/index.ts | 6 | CRITICAL | Importing .ts extension in ESM/Node runtime may fail | Use .js extension to match built output and existing pattern |
| src/core/utils/runtime-config.ts | 1–3 | CRITICAL | Imports use .ts extensions; inconsistent with repo’s .js imports | Switch to .js extensions for runtime imports and type-only imports |
| src/core/utils/runtime-config.ts | ~79 | IMPORTANT | `LLM_PROVIDER` env var not honored (docs claim it is) | Read and validate `process.env.LLM_PROVIDER` before app config |
| src/core/utils/runtime-config.ts | ~155–161 | IMPORTANT | `REDIS_DB` parsing inconsistent; may allow negatives/NaN | Reuse `parseNumber` and preserve undefined fallback |
| src/core/utils/runtime-config.ts | ~189 | IMPORTANT | Port parsing permits floats/out-of-range | Validate integer 0–65535, else fail back to default |
| src/index.ts | ~13–16 | CRITICAL | Missing early validation of required GitHub App secrets | Fail fast with clear error before initializing Probot |
| src/core/utils/runtime-config.ts | ~214, ~238 | IMPORTANT | Public functions lack docstrings (guideline 4) | Add concise JSDoc for exported functions |

Additionally, please add tests for precedence (env > config > defaults), boolean parsing (including invalid values), and boundary checks (ports, Redis DB). This is critical code, so aim for high coverage per guidelines.

File: src/core/utils/runtime-config.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+  const authzFallbackToRepo = parseBooleanEnv(
+    process.env.AUTHZ_FALLBACK_TO_REPO,
+    app.authzFallbackToRepo ?? false
+  )
+
+  const gitPath = process.env.GIT_PATH || app.gitPath || '/usr/bin/git'
+
+  const host = process.env.HOST || app.host || '0.0.0.0'
+
+  const port = parseNumber(process.env.PORT, app.port ?? 3000)
```
Comment (revu-bot-dev) [inline src/core/utils/runtime-config.ts]: ### [IMPORTANT] Port parsing should validate integer range

- Why it matters:
  - `parseNumber` permits floats and very large numbers; `express.listen` expects a valid TCP port (integer 0–65535).
  - Invalid ports cause confusing runtime failures.
- Proposed change: validate that `PORT` is an integer within [0, 65535]; otherwise, fall back to a sane default.


```suggestion
  const port = (() => {
    const raw = process.env.PORT ?? String(app.port ?? 3000)
    const n = Number(raw)
    return Number.isInteger(n) && n >= 0 && n <= 65535 ? n : 3000
  })()
```

File: src/core/utils/runtime-config.ts [hunk 2]
```diff
@@ -0,0 +1,251 @@
+import { getAppConfig } from './config-loader.ts'
+import type { RevuAppConfig } from '../../types/config.ts'
+import { logSystemWarning } from '../../utils/logger.ts'
```
[the remark this answers is not in the document]
Comment (devthejo) [reply src/core/utils/runtime-config.ts]: We use experimental node ts run, we don't build the code, the extensions .ts are fine here

File: src/core/utils/runtime-config.ts [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+  logSystemWarning(
+    new Error('Invalid boolean env var, expected "true" or "false"'),
+    { context_msg: `value="${raw}"` }
+  )
+  return fallback
+}
+
+function buildRuntimeConfig(app: RevuAppConfig): RuntimeConfig {
+  // LLM / models
+  const provider: 'anthropic' | 'openai' = app.llmProvider ?? 'anthropic'
```
[the remark this answers is not in the document]
Comment (revu-bot-dev) [reply src/core/utils/runtime-config.ts]: I could not generate a confident, useful automated reply for this discussion message. Please clarify your question or add more context if you would like a more detailed follow-up.

[the remark this answers is not in the document]
Comment (devthejo) [reply src/core/utils/runtime-config.ts]: I want to keep app as the only source of truth, any other suggestion to fix this ?

**Review by revu-bot-dev — COMMENTED** [review]
An error occurred: 400 Invalid schema for response_format 'code_review': In context=(), 'additionalProperties' is required to be supplied and to be false.

[Revu logs](https://grafana.fabrique.social.gouv.fr/explore?schemaVersion=1&panes=%7B%22y1w%22%3A%7B%22datasource%22%3A%22P8E80F9AEF21F6940%22%2C%22queries%22%3A%5B%7B%22refId%22%3A%22A%22%2C%22expr%22%3A%22%7Bcluster%3D%5C%22ovh-prod%5C%22%2C%20namespace%3D%5C%22revu%5C%22%2C%20container%3D%5C%22app%5C%22%7D%20%7C%20json%20%7C%20level%20%3D%20%60error%60%22%2C%22queryType%22%3A%22range%22%2C%22datasource%22%3A%7B%22type%22%3A%22loki%22%2C%22uid%22%3A%22P8E80F9AEF21F6940%22%7D%2C%22editorMode%22%3A%22builder%22%2C%22direction%22%3A%22forward%22%7D%5D%2C%22range%22%3A%7B%22from%22%3A%221763976362653%22%2C%22to%22%3A%221763976962653%22%7D%7D%7D&orgId=1)

**Review by revu-bot-dev — COMMENTED** [review]
### Overall PR Summary

This refactor centralizes environment/config handling into a `RuntimeConfig` layer and updates senders/handlers to consume it. The direction is solid and improves consistency and testability. Most changes are straightforward wiring, but there are a couple of behavioral shifts and one subtle testing/initialization concern worth addressing.

Key points:

- [IMPORTANT] `getAppConfig` no longer validates `LLM_PROVIDER`, but tests still expect a warning on invalid env; that behavior moved to `runtime-config` and is now untested.
- [IMPORTANT] `rate-limit.ts` and `retry.ts` capture `runtime` at module load via `getRuntimeConfigSync`, which can diverge from the async-initialized runtime (and from `config.json`) in some startup orders.
- [MINOR] `build-discussion-prompt-segments` now hardcodes `DEFAULT_MAX_CHARS_PER_FILE` and relies on the caller to pass runtime values; this is fine but could use a small comment to avoid future drift.

Addressing these will keep behavior consistent and avoid hard-to-debug config discrepancies.

File: src/core/utils/runtime-config.ts [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+  if (!cachedRuntimeConfig) {
+    // Fallback to building a runtime config from defaults + env only.
+    // This avoids requiring an async initialization step in purely
+    // synchronous modules. Config from config.json will not be
+    // reflected in this rare path.
+    const appConfig: RevuAppConfig = { ...DEFAULT_APP_CONFIG }
+    cachedRuntimeConfig = buildRuntimeConfig(appConfig)
+  }
+  return cachedRuntimeConfig
+}
```
[the remark this answers is not in the document]
Comment (devthejo) [reply src/core/utils/runtime-config.ts]: Maybe we should do DI for config, WDYT ?
